webapp/channels/src/packages/mattermost-redux/src/reducers/entities/threads/counts.ts

```diff
@@ -15,6 +15,31 @@ function isDmGmChannel(channelType: Channel['type']) {
     return channelType === General.DM_CHANNEL || channelType === General.GM_CHANNEL;
 }
 
+function handleChannelThreadCounts(state: ThreadsState['countsInChannel'], action: GenericAction, key: keyof ThreadsState['countsInChannel'][keyof ThreadsState['countsInChannel']]) {
+    const {total, channel_id: channelId} = action.data;
+
+    if (state?.[channelId]?.[key] === total) {
+        return state;
+    }
+
+    if (!state[channelId]) {
```

Comment:
@hmhealey it will fail if you try to spread a non-object aka `undefined`.

Reply:
Actually, I think this case works fine. Oddly enough `...undefined` will cause an error, but if you spread an variable with a value of `undefined`, it just returns an empty object. 0/5 on actually changing it though since I know I had to double check that behaviour myself, so it's not really intuitive anyway